internal/cloud/backend_apply_test.go

```diff
@@ -1566,7 +1566,7 @@ func TestCloud_applyVersionCheck(t *testing.T) {
 				hasRemote := strings.Contains(output, "Running apply in Terraform Cloud")
 				hasSummary := strings.Contains(output, "1 added, 0 changed, 0 destroyed")
 				hasResources := run.State.HasManagedResourceInstanceObjects()
-				if !tc.forceLocal && tc.hasOperations {
+				if !tc.forceLocal && tc.executionMode == "remote" {
```

Comment:
```suggestion
				if !tc.forceLocal && !isLocalExecutionMode(tc.executionMode) {
```

Whatcha think about using the same function here in the tests? Reason being: I think to be accurate, you'd want to make sure the `"agent"` execution mode also results in remote operations, and comparing to "remote" would miss that.

Reply:
Yes, this makes complete sense. I'll commit this suggestion :)